### PR TITLE
Extend Virtual HTLC Operations

### PR DESCRIPTION
**Pull request by crypto-ape**
Hey Monkeys!

This PR is associated with #1887 .

I have extended the `htlc_redeemed_operation` with the preimage information and refactored other fields carrying information form the _htlc object_ to match them with those in the `htlc_refund_operation`.

@abitmore mentioned _time_ in the issue, but I have not included the expiration time in the operations. If anybody deems the inclusion of this information to be of any use, please express it, so we can discuss it and/or add it.

There is also a small refactor I also mentioned in the issue, please review it carefully.

Ape out!

### CODE REVIEW

**Comment by abitmore** [comment]
Renaming existing fields is a bad idea IMHO.

**Comment by crypto-ape** [comment]
I won't disagree, no plan to force it. It is just a proposition to be reviewed / discussed.

The point is that it would be much more coherent from a programmer's perspective.

Alas, such refactors are more than unfortunate. Yet, it is a relatively new code and, the question is, will it actually break anything anywhere? And even if so, isn't it worth it?

**Comment by crypto-ape** [comment]
The renames are reverted.
The branch is freshly rebased.

File: libraries/protocol/include/graphene/protocol/htlc.hpp [hunk 1]
```diff
@@ -120,8 +120,10 @@ namespace graphene { namespace protocol {
          struct fee_parameters_type {};
 
          htlc_redeemed_operation() {}
-         htlc_redeemed_operation( htlc_id_type htlc_id, account_id_type from, account_id_type to, account_id_type redeemer, asset amount ) :
-               htlc_id(htlc_id), from(from), to(to), redeemer(redeemer), amount(amount) {}
+         htlc_redeemed_operation( htlc_id_type htlc_id, account_id_type from, account_id_type to, account_id_type redeemer, asset amount,
```
Comment (abitmore) [inline libraries/protocol/include/graphene/protocol/htlc.hpp]: Please wrap this line (max = 117 chars).

File: libraries/protocol/include/graphene/protocol/htlc.hpp [hunk 2]
```diff
@@ -173,18 +178,26 @@ namespace graphene { namespace protocol {
          struct fee_parameters_type {};
 
          htlc_refund_operation(){}
-         htlc_refund_operation( const htlc_id_type& htlc_id, const account_id_type& to ) 
-         : htlc_id(htlc_id), to(to) {}
+         htlc_refund_operation( const htlc_id_type& htlc_id, const account_id_type& htlc_from, const account_id_type& htlc_to,
```
Comment (abitmore) [inline libraries/protocol/include/graphene/protocol/htlc.hpp]: Please wrap this line.

File: libraries/protocol/include/graphene/protocol/htlc.hpp (hunk 1 above)
Comment (crypto-ape) [reply libraries/protocol/include/graphene/protocol/htlc.hpp]: Done.

File: libraries/protocol/include/graphene/protocol/htlc.hpp (hunk 2 above)
Comment (crypto-ape) [reply libraries/protocol/include/graphene/protocol/htlc.hpp]: Done.

**Comment by abitmore** [comment]
Please bump DB_VERSION to current date (https://github.com/crypto-ape/bitshares-core/blob/extend_virtual_htlc_operations/libraries/chain/include/graphene/chain/config.hpp#L330).

**Comment by crypto-ape** [comment]
Bumped.

**Pull request merged**